### ISSUE TITLE
Make -j2 on RPi3 and Raspbian Jessy failed

### ISSUE THREAD

**Issue by mariodian** [bitcoin/bitcoin#9099]
### Describe the issue
I tried to compile 0.13.1 on my Raspberry Pi3 but it failed on "make". I have all the dependencies installed because I've run 0.12.1 (pruned) before with no issues.

#### If so, please list the steps to reproduce below:
```
git clone -b 0.13 https://github.com/bitcoin/bitcoin.git
cd bitcoin
./autogen.sh
./configure --enable-upnp-default --disable-wallet
make -j2
```

### Actual behaviour
I got following errors:

```
CXXLD    bitcoind
bitcoind-bitcoind.o: In function `WaitForShutdown(boost::thread_group*)':
/home/pi/bin/bitcoin/src/bitcoind.cpp:47: undefined reference to `ShutdownRequested()'
/home/pi/bin/bitcoin/src/bitcoind.cpp:52: undefined reference to `ShutdownRequested()'
/home/pi/bin/bitcoin/src/bitcoind.cpp:56: undefined reference to `Interrupt(boost::thread_group&)'
bitcoind-bitcoind.o: In function `AppInit(int, char**)':
/home/pi/bin/bitcoin/src/bitcoind.cpp:92: undefined reference to `HelpMessage(HelpMessageMode)'
/home/pi/bin/bitcoin/src/bitcoind.cpp:85: undefined reference to `LicenseInfo()'
/home/pi/bin/bitcoin/src/bitcoind.cpp:159: undefined reference to `InitLogging()'
/home/pi/bin/bitcoin/src/bitcoind.cpp:160: undefined reference to `InitParameterInteraction()'
/home/pi/bin/bitcoin/src/bitcoind.cpp:161: undefined reference to `AppInit2(boost::thread_group&, CScheduler&)'
/home/pi/bin/bitcoin/src/bitcoind.cpp:178: undefined reference to `Shutdown()'
/home/pi/bin/bitcoin/src/bitcoind.cpp:171: undefined reference to `Interrupt(boost::thread_group&)'
bitcoind-bitcoind.o: In function `main':
/home/pi/bin/bitcoin/src/bitcoind.cpp:188: undefined reference to `noui_connect()'
collect2: error: ld returned 1 exit status
Makefile:3360: recipe for target 'bitcoind' failed
make[2]: *** [bitcoind] Error 1
make[2]: Leaving directory '/home/pi/bin/bitcoin/src'
Makefile:8381: recipe for target 'install-recursive' failed
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory '/home/pi/bin/bitcoin/src'
Makefile:672: recipe for target 'install-recursive' failed
make: *** [install-recursive] Error 1
```

### What version of bitcoin-core are you using?
0.13.1 https://github.com/bitcoin/bitcoin/commit/59ce53c13cfca85418fad039538acb3bd21bbcf4

### Machine specs:
- OS: Raspbian Jessy
- CPU: 1.2 GHZ quad-core ARM Cortex A53
- RAM: 1GB
- Disk size: 8GB
- Disk Type (HD/SDD): MicroSD

### Any extra information that might be useful in the debugging process.
Not sure if it's related, but I had to enable 1GB swap otherwise I'd run out of memory on:

```
CXX      libbitcoin_server_a-main.o
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-4.9/README.Bugs> for instructions.
Makefile:5083: recipe for target 'libbitcoin_server_a-main.o' failed
make[2]: *** [libbitcoin_server_a-main.o] Error 4
make[2]: Leaving directory '/home/pi/bin/bitcoin/src'
Makefile:8357: recipe for target 'all-recursive' failed
make[1]: *** [all-recursive] Error 1
make[1]: Leaving directory '/home/pi/bin/bitcoin/src'
Makefile:668: recipe for target 'all-recursive' failed
make: *** [all-recursive] Error 1
```

**Comment by sipa** [comment]
Don't compile with -j2. The system doesn't have nearly enough memory to run to compiler instances in parallel. It hardly has enough for one.

**Comment by mariodian** [comment]
I'm not sure whether -j2 is related to the above issue anyway because I've compiled 0.12.1 with -j2 before.

**Comment by sipa** [comment]
I believe memory usage for compiling has gone up in recent versions. Your compiler is OOMing. Don't run two instances of it simultaneously.

**Comment by sipa** [comment]
Closing as duplicate of #6658.

**Comment by AlonzoRicardo** [comment]
How does reducing the value of `make -j<value>` solve the `undefined reference` errors? 
It only solves the `g++: internal compiler error: Killed (program cc1plus)` error

**Comment by MarcoFalke** [comment]
`make distclean` should solve those (you will need to run `./configure` again)
